packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -81,8 +82,9 @@ export const ElectionProgressBar = (props: ElectionProgressBarProps) => {
   const constants = useCouncilConstants()
 
   const [inactiveEndBlock, announcingEndBlock, votingEndBlock, revealingEndBlock] = periodInformation?.periodEnds ?? []
-  const endDates = periodInformation?.periodEnds.map((block) => blockToDate(block - currentBlock))
+  const endDates = periodInformation?.periodEnds.map((block) => blockToDate(block))
```

Comment:
Instead I think this is good enough:
```suggestion
  const endDates = useMemo(() => periodInformation?.periodEnds.map((block) => blockToDate(block - currentBlock)), [props.electionStage])
```
When the page is reloaded the time won't be exactly the same but it will actually be more accurate.

Reply:
I have tried to use props.electionStage, at that time sometimes the endDate is calculated as undefined, so I changed it to periodInformation.currentStage.